fedimint-testing/src/gateway.rs

```diff
@@ -60,6 +60,7 @@ impl GatewayTest {
     /// Note: this makes the database opened, which can lead to gateway
     /// wanting to rejoin it. Better get your own client instead of being
     /// lazy here.
+    // TODO: This stuff is horrible, why do we need it?
```

Comment:
Yes agreed it is horrible, its been a long standing bandaid. The reason it is currently needed is that we use this hack to verify the gateway's state machines, since otherwise the client would automatically call them via the REST API.

I think we have two options to try to fix it.

1) Remove test coverage over the gateway's state machines and just verify the correct operation happened from the client side.
2) Figure out a different workaround to allow verification of the gateway's state machines. Perhaps we could wrap the client REST client and not issue any REST requests when we're in "testing" mode?

Reply:
@m1sterc001guy IMO the errors and/or other metadata passed all the way to client should be enough to verify what happened.

Reply:
The gateway intentionally doesn't pass back a ton of info to the client about an error that happens though, since it could leak info about the gateway's lightning node for example. Ideally I'd like to keep the coverage of the gateway's state machines but I'd need to look closer if we already have redundant checks in other spots. Maybe we can make an issue an differ this issue to after this PR